pkg/pillar/cmd/zedagent/hardwareinfo.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/lf-edge/eve/api/go/info"
+	"github.com/lf-edge/eve/pkg/pillar/hardware"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/zedcloud"
+)
+
+func hardwareInfoTask(ctxPtr *zedagentContext, triggerHwInfo <-chan struct{}) {
+	wdName := agentName + "hwinfo"
+
+	stillRunning := time.NewTicker(30 * time.Second)
+	ctxPtr.ps.StillRunning(wdName, warningTime, errorTime)
+	ctxPtr.ps.RegisterFileWatchdog(wdName)
+
+	for {
+		select {
+		case <-triggerHwInfo:
+			start := time.Now()
+			log.Function("HardwareInfoTask got message")
+
+			PublishHardwareInfoToZedCloud(ctxPtr)
+			ctxPtr.iteration++
+			log.Function("HardwareInfoTask done with message")
+			ctxPtr.ps.CheckMaxTimeTopic(wdName, "PublishHardwareInfo", start,
+				warningTime, errorTime)
+		case <-stillRunning.C:
+		}
+		ctxPtr.ps.StillRunning(wdName, warningTime, errorTime)
+	}
+}
+
+//PublishHardwareInfoToZedCloud send HardwareInfo message
+func PublishHardwareInfoToZedCloud(ctx *zedagentContext) {
+	deviceUUID := devUUID.String() + "hwinfo"
+	hwInfo := new(info.HardwareInfo)
+	bailOnHTTPErr := false // sended not so ofen
+
+	// Get information about disks
+	disksInfo, err := hardware.ReadSMARTinfoForDisks()
+	if err == nil {
```

Comment:
Seems like even if ReadSMARTinfoForDisks fail you send an info message, which is basically empty. I suspect it is better to log an error and send nothing. 
In such a case the code becomes if err != nil { ... return}

Reply:
Corrected